src/tests/ov_helpers/ov_snippets_models/src/subgraph_mlp.cpp

```diff
@@ -0,0 +1,296 @@
+    }
+    auto add = std::make_shared<ov::op::v0::Constant>(ov::element::f32,
+                                                      input_shapes[0].to_shape(),
+                                                      std::vector<float>{0.1122});
+
+    ov::builder::subgraph::FakeQuantizeOnData onData =
+        {256, {1, 1}, {0.f}, {2.55f}, {0.f}, {255.f}, ov::element::u8};
+    std::shared_ptr<Node> current = A;
+
+    for (size_t mm_count = 0; mm_count < num_layers; ++mm_count) {
+        current = ov::builder::subgraph::makeFakeQuantizeTypeRelaxed(current, ov::element::f32, onData);
+        auto B = std::make_shared<ov::op::v0::Constant>(ov::element::i8, input_shapes[0].to_shape(), std::vector<float>{0.1122f + mm_count});
+
+        current = std::make_shared<op::TypeRelaxed<ov::op::v0::MatMul>>(
+            std::vector<element::Type>{element::f32, element::f32},
+            std::vector<element::Type>{element::f32},
+            ov::op::TemporaryReplaceOutputType(current, element::f32).get(),
+            ov::op::TemporaryReplaceOutputType(B, element::f32).get(),
+            false,
+            true);
+
+        current = std::make_shared<op::TypeRelaxed<ov::op::v1::Multiply>>(
+            std::vector<element::Type>{element::f32, element::f32},
+            std::vector<element::Type>{element::f32},
+            ov::op::TemporaryReplaceOutputType(current, element::f32).get(),
+            ov::op::TemporaryReplaceOutputType(add, element::f32).get());
+        for (size_t i = 0; i < 2; ++i) {
+            auto constant = std::make_shared<ov::op::v0::Constant>(ov::element::f32,
+                                                                   ov::Shape{input_shapes[0].to_shape()[0]},
+                                                                   std::vector<float>{0.1122f + i});
+            current = std::make_shared<op::TypeRelaxed<ov::op::v1::Add>>(
+                std::vector<element::Type>{element::f32, element::f32},
+                std::vector<element::Type>{element::f32},
+                ov::op::TemporaryReplaceOutputType(current, element::f32).get(),
+                ov::op::TemporaryReplaceOutputType(constant, element::f32).get());
+        }
+    }
+    ov::builder::subgraph::FakeQuantizeOnData onData2 =
+        {256, {1, 1}, {0.f}, {2.55f}, {0.f}, {255.f}, ov::element::f32};
+    current = ov::builder::subgraph::makeFakeQuantizeTypeRelaxed(current, ov::element::f32, onData2);
+    auto softmax = std::make_shared<ov::op::v8::Softmax>(current, 1);
+    ov::builder::subgraph::FakeQuantizeOnData onData1 =
+        {256, {1, 1}, {0.f}, {2.55f}, {0.f}, {255.f}, ov::element::f32};
+    auto dq_A = ov::builder::subgraph::makeFakeQuantizeTypeRelaxed(softmax, ov::element::f32, onData1);
+
+    auto result = std::make_shared<ov::op::v0::Result>(dq_A);
+    return std::make_shared<Model>(ResultVector{result}, ParameterVector{A_param});
+}
+
+std::shared_ptr<ov::Model> MLPSeqTypeRelaxedFunction::initReference() const {
```

Comment:
Minor: could you please correct order of method bodies here to align with class definitions in header? It will be convenient to read the code

Reply:
Done